File: benchmarks/pandas/bench_dataframe_sort.py
"""Benchmark: dataframe_sort — sort a 100k-row DataFrame by two columns"""
import json, time
import numpy as np
import pandas as pd

ROWS = 100_000
WARMUP = 3
ITERATIONS = 10

rng = np.random.default_rng(42)
a = [f"group_{i % 100}" for i in range(ROWS)]
b = rng.random(ROWS) * 1000
df = pd.DataFrame({"a": a, "b": b})

for _ in range(WARMUP):
    df.sort_values(["a", "b"])

start = time.perf_counter()
for _ in range(ITERATIONS):
    df.sort_values(["a", "b"])
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "dataframe_sort",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))

File: benchmarks/tsb/bench_dataframe_sort.ts
/**
 * Benchmark: dataframe_sort — sort a 100k-row DataFrame by two columns
 */
import { DataFrame } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;

const a = Array.from({ length: ROWS }, (_, i) => `group_${i % 100}`);
const b = Float64Array.from({ length: ROWS }, () => Math.random() * 1000);
const df = new DataFrame({ a, b });

for (let i = 0; i < WARMUP; i++) {
  df.sort_values(["a", "b"]);
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  df.sort_values(["a", "b"]);
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "dataframe_sort",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);
